src/main/java/com/karl/http/HTTPHandler.java

```diff
@@ -10,12 +10,30 @@ public HTTPHandler(Connectable connection) {
     this.connection = connection;
   }
 
+  public String createRequestString(HTTPStatus status, String headers, String body) {
```

Comment:
Method overloads would help here so that you have one function that would not have Body as a function param, so you can call that method for a GET, instead of having to pass empty string like this `createRequestString(HTTPStatus.Ok, "", "")`. 
Same for the headers param. 
This would improve readability and usage.